### PR TITLE
Detect region correctly in Local Zones

### PR DESCRIPTION
**Pull request by nmx**
The previous logic assumed that the region could always be determined
by stripping the last character from the Availability Zone. This is no
longer true; for example the Local Zone us-west-2-lax-1a corresponds
to the region us-west-2.

Read the region directly from the instance identity document instead.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_confighelper.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from cloudwatch.modules.configuration.confighelper import ConfigHelper
 from cloudwatch.modules.configuration.metadatareader import MetadataReader
 from helpers.fake_http_server import FakeServer
-
+from helpers.fake_metadata import FAKE_REGION, FAKE_IDENTITY_DOCUMENT_STRING
```
Comment (hdj630) [inline test/test_confighelper.py]: fake_metadata is missing in the PR.

Comment (nmx) [reply test/test_confighelper.py]: Updated

File: test/test_confighelper.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def test_hostname_loaded_from_file_is_not_overriden_by_metadata(self):
     def test_instance_id_is_used_as_hostname_if_not_specified_in_config(self):
         expected_host = "Valid_Instance_ID"
         self.server.set_expected_response(expected_host, 200)
-        self.config_helper = ConfigHelper(config_path=ConfigHelperTest.VALID_CONFIG_WITH_CREDS_ONLY, metadata_server=self.server.get_url())
+        self.config_helper = ConfigHelper(config_path=ConfigHelperTest.VALID_CONFIG_WITH_CREDS_AND_REGION, metadata_server=self.server.get_url())
```
Comment (hdj630) [inline test/test_confighelper.py]: What's the purpose of this change?

Comment (nmx) [reply test/test_confighelper.py]: To be honest I wasn't sure how this test passed before. Without providing the region, this results in two calls to the metadata service, but we are only mocking the result of one of them.

Comment (hdj630) [reply test/test_confighelper.py]: I see, sounds good.

Comment (nmx) [reply test/test_confighelper.py]: Ah, I believe I know why it worked - before, it would just truncate the hostname as if it were the AZ string. With my change, however, it tried to parse the hostname as JSON, which fails. That exception must get suppressed and the region is treated as an empty string.

I think this change is necessary unless we enhance the fake metadata service to be able to mock different responses to different URLs (or a sequence of responses).
